drivers/dma/dma_stm32.c

```diff
@@ -36,6 +99,24 @@ static void dma_stm32_clear_stream_irq(struct device *dev, u32_t id)
 	func_ll_clear_tc[id](dma);
 	func_ll_clear_ht[id](dma);
 	stm32_dma_clear_stream_irq(dma, id);
+
+#if defined(CONFIG_DMA_STM32_V2) && defined(CONFIG_DMAMUX_STM32)
+	/* also clear the corresponding dmamux irq */
+	struct device *dev_mux =
+		 device_get_binding((const char *)"DMAMUX_1");
+
+	const struct dmamux_stm32_config *devmux_config =
+					dev_mux->config->config_info;
+	/* retrieve the dma channel from the dmamux channel */
+	u32_t mux_id = ((strcmp(dev->config->name, "DMA_1") == 0) ? id :
+	 ((devmux_config->channel_nb / 2) + id));
```

Comment:
This should be explained

Reply:
this formula is used to retrieve the dmamux channel from the dma instance + channel (stream)
put in a macro